server/monitor/src/main/java/org/apache/accumulo/monitor/rest/tables/TablesResource.java

```diff
@@ -0,0 +1,232 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.client.impl.Namespaces;
+import org.apache.accumulo.core.client.impl.Tables;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.impl.KeyExtent;
+import org.apache.accumulo.core.master.thrift.TableInfo;
+import org.apache.accumulo.core.master.thrift.TabletServerStatus;
+import org.apache.accumulo.core.metadata.MetadataTable;
+import org.apache.accumulo.core.metadata.RootTable;
+import org.apache.accumulo.monitor.Monitor;
+import org.apache.accumulo.monitor.rest.tservers.TabletServer;
+import org.apache.accumulo.monitor.rest.tservers.TabletServers;
+import org.apache.accumulo.server.client.HdfsZooInstance;
+import org.apache.accumulo.server.master.state.MetaDataTableScanner;
+import org.apache.accumulo.server.master.state.TabletLocationState;
+import org.apache.accumulo.server.tables.TableManager;
+import org.apache.accumulo.server.util.TableInfoUtil;
+import org.apache.hadoop.io.Text;
+
+/**
+ *
+ * Generates a tables list from the Monitor as a JSON object
+ *
+ * @since 2.0.0
+ *
+ */
+@Path("/tables")
+@Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+public class TablesResource {
+
+  private static final TabletServerStatus NO_STATUS = new TabletServerStatus();
+
+  /**
+   * Generates a table list based on the namespace
+   *
+   * @param namespace
+   *          Namespace used to filter the tables
+   * @return Table list
+   */
+  private TablesList generateTables(String namespace) {
+    Instance inst = Monitor.getContext().getInstance();
+    Map<String,String> tidToNameMap = Tables.getIdToNameMap(inst);
+    SortedMap<String,TableInfo> tableStats = new TreeMap<>();
+
+    if (Monitor.getMmi() != null && Monitor.getMmi().tableMap != null)
+      for (Entry<String,TableInfo> te : Monitor.getMmi().tableMap.entrySet())
+        tableStats.put(Tables.getPrintableTableNameFromId(tidToNameMap, te.getKey()), te.getValue());
+
+    Map<String,Double> compactingByTable = TableInfoUtil.summarizeTableStats(Monitor.getMmi());
+    TableManager tableManager = TableManager.getInstance();
+
+    SortedMap<String,String> namespaces = Namespaces.getNameToIdMap(Monitor.getContext().getInstance());
+
+    TablesList tableNamespace = new TablesList();
+    List<TableInformation> tables = new ArrayList<>();
+
+    // Add the tables that have the selected namespace
+    for (String key : namespaces.keySet()) {
+      if (namespace.equals("*") || namespace.equals(key) || (key.equals("") && namespace.equals("-"))) {
```

Comment:
I've previously tossed around the idea of creating a proper name for the default namespace, and leaving empty string as an alias. Cleaning up the namespace utility classes to do this is probably a bit tedious. If it helps with this interface, it may help prioritize giving it a real name.